### ISSUE TITLE
Quiver step failing since it can not find some files to delete

### ISSUE THREAD

**Issue by MichaelsGITIGIT** [PacificBiosciences/FALCON_unzip#31]
I got to the last step of falcon unzip; quivering...

I converted RSII bax.h5 files to bam. The command was the following:
`/opt/pacbio/smrtlink/install/smrtlink-fromsrc_3.0.5.175021,175083-175021-174993-174993/bundles/smrttools/install/smrttools-fromsrc_3.0.5.175083/smrtcmds/bin/bax2bam m1.bax.h5 m2.bax.h5 m3.bax.h5 -o ./outp/converted --subread`

Then I provided all bam files (`*.subreads.bam`) via input_bam.fofn.

But I get the following error:
`rm: cannot remove './4-quiver/cns_output/cns_p_ctg.fasta': No such file or directory
rm: cannot remove './4-quiver/cns_output/cns_p_ctg.fastq': No such file or directory
rm: cannot remove './4-quiver/cns_output/cns_h_ctg.fasta': No such file or directory
rm: cannot remove './4-quiver/cns_output/cns_h_ctg.fastq': No such file or directory
`

**Comment by pb-jchin** [comment]
Hi, there is nothing to worry about that. The code is designed to be re-runnable. To make sure we get "clean" results, in some stage the previously generated files should be removed. I could use some "test" command in "bash" to avoid the messages. It was not in the critical path for solving the main problem so I did ignore that. We should certainly improve it later.

**Comment by MichaelsGITIGIT** [comment]
Oh, sorry, I did not specify my question well enough; The problem is, that pipeline stops after this message, but all the folders in `4-quiver` are emtpy.

**Comment by pb-jchin** [comment]
My guess is that you might not have the 'input_bam.fofn' pointing to the right bam files. (Sorry for somehow poor usability for now. I typically focus on algorithms and bioinformatics analysis first.)

**Comment by MichaelsGITIGIT** [comment]
No worries about the usability! I fully understand that there is first focus on the algorithm.

Just to make sure; I have to provide the `subread` bam files. NOT the `scraps` output, right?

**Comment by pb-jchin** [comment]
in the latest code, you need to have a file "input_bam.fofn" and set it with the `fc_unzip.cfg` ( https://github.com/PacificBiosciences/FALCON_unzip/blob/master/examples/fc_unzip.cfg ) so the code knows where to the get the "bam" files. 

In the latest code, check `https://github.com/PacificBiosciences/FALCON_unzip/blob/master/examples/unzip.sh` to see how to call it. 

In the earlier code, check
"https://github.com/PacificBiosciences/FALCON_unzip/blob/master/examples/unzip_obsoleted.sh" for the whole example. (You might to comment out some lines if those steps have be executed.)

**Comment by MichaelsGITIGIT** [comment]
Hmmm... I did it the way you just described (at least that was my plan). But BAM files are only necessary for last step (`4-quiver`), right?

It is also strange that I do not get any error, but it just finishes without result.

Thanks!

**Comment by pb-jchin** [comment]
the bam files are only need for the `4-quiver`.  Check if the `3-unzip/read_maps` directory to see if the read track information is generated. 

if it is empty, check if you have these files in `3-unzip`,
```
all_h_ctg_ids
all_p_ctg_edges
all_h_ctg_edges
all_p_ctg.fa
all_h_ctg.fa
all_phased_reads
``` 

If there is, then check if you have file insides `4-quiver/reads`

You will need to have `samtools` in your path for the quvier part.

**Comment by MichaelsGITIGIT** [comment]
I again run the assembly from scratch, making sure that none of my data is strange/corrupt.

Still only empty folders in `4-quiver`. But I have all the files you mentioned above `all_h_ctg_ids, all_p_ctg_edges`, etc.) and they are not empty.

So maybe it is a missing/old tool. Which versions of samtools, quiver, pbalign, variantcaller do u use? and what about pysam?

Thanks a lot!

**Comment by pb-jchin** [comment]
(1) check if you have `./3-unzip/1-hasm/hasm_done`
(2) if you have it, the code will trace reads to each haplotigs, after tracing, you should get a file called `track_reads_h_done`, you should have reads (in SAM format) in `4-quiver/reads/` 
Can you check the content in `4-quiver/reads/`? If you don't have it, it is likely the bam files are not correctly referenced.

here is the version related informati
```
pysam >= 0.8.4
```
samtools is tested with 0.1.17/1.2/1.3.1, we only use some generic functions (converting SAM to BAM, sort, indexing) 

quiver, variantcaller are from SMRTanalysis 3.0.0 or 3.1.0

**Comment by MichaelsGITIGIT** [comment]
Still not working. To answer your questions:
(1) yes, I have `./3-unzip/1-hasm/hasm_done`
(2) Yes I have the file `track_reads_h_done`. But I have NO data in `4-quiver/reads/`. I get a lot of folders in there but they are all empty...

The paths for the BAM-files are correct. Because if I missspell one BAM-File deliberately falcon_unzip complains.

**Comment by MichaelsGITIGIT** [comment]
Solved now. To save space we used to delete the files from the first two steps (`0-rawreads`, `1-preads_ovl`) since we thought they are not needed by the last `quiver `step. Now we increased the space on the machine and did a run (from scratch) without deleting it... Et voilà everything worked.

I do not know the exact cause of the problem, but it seems it is better to keep everything in place until the whole pipeline has run through.
